### ISSUE TITLE
why dotspatial not use GDAL's BuildOverviews function to create pyramids?

### ISSUE THREAD

**Issue by wangchao5032** [DotSpatial/DotSpatial#940]
I read the codes and found a question :when Dotspatial add a big tiff image (for example a 2G tiff), it will create pyramids file itself but not use the GDAL's BuildOverviews function. Now that Dotspatial has used GDAL to read tiff image, why not use GDAL's existing function to create pyramids so that to add the big image? And GDAL's BuildOverviews function is more quickly.

**Comment by jany-tenaj** [comment]
@wangchao5032 That's a good question.

I've seen this, too, while trying to fix some other problems inside GdalImage. As long as we get BuildOverviews to work I think it would be a good idea to use it instead.

**Comment by wangchao5032** [comment]
Thank you for your answer! I am so exited! I think the difficulty for Dotspatial to use GDAL's pyramid file is: Dotspatial has its own class and interface, so GDAL's data may have some problems to fit the Dotspatial? (I am a newcomer so maybe my question is a little naive......)

**Comment by wangchao5032** [comment]
@jany-tenaj 
Thank you for your answer! I am so exited! I think the difficulty for Dotspatial to use GDAL's pyramid file is: Dotspatial has its own class and interface, so GDAL's data may have some problems to fit the Dotspatial? (I am a newcomer so maybe my question is a little naive......)

**Comment by jany-tenaj** [comment]
@wangchao5032 If the image that is loaded already has pyramids (build by GDAL or something else) those are used by default. It's only when there are no pyramids that DS builds its own.

**Comment by wangchao5032** [comment]
@jany-tenaj 
I agree with you in GDAL's way to read pyramid file. But the Dotspatial seems only to know its own pyramid file format: .mwi and .mwh(in MapRsterLayer.cs). And if the folder lost .mwi or .mwh file, Dotspatial will create pyramid and it will take a long time. Dotspatial seems only open image file by GDAL's Open function first and then handles the image in its own way include how to create and identify pyramid files and not rely on GDAL again.
The talk with you helps me to think a lot! I will try to solve the problem! Thanks very much! I am a student form Zhejiang University in China and major in GIS. Can I ask you the next time I have a problem? I might be a little presumptuous...

**Comment by jany-tenaj** [comment]
As I said, DS already uses internal GDAL overviews if they exist. [See line 278 and following](https://github.com/DotSpatial/DotSpatial/blob/master/Source/DotSpatial.Data.Rasters.GdalExtension/GdalImage.cs#L278).

We only have to change [line 236 and following](https://github.com/DotSpatial/DotSpatial/blob/master/Source/DotSpatial.Data.Rasters.GdalExtension/GdalImageProvider.cs#L236) to make DS create GDAL overviews instead of DS pyramids.

**Comment by wangchao5032** [comment]
@jany-tenaj 
I think i have got your idea roughly. But when i add a tiff image and debug it, the code cannot jump to the GdalImageProvider's OpenFile function, and it jumped to the MapRasterLayerer's constructor function in fact, and just check the .mwi and .mwh file but don't check if exits a GDAL's pyramid file...... Maybe i debuged  by mistake in some places or some places in the code haven't consider the GDAL's pyramid file carelessly? 
I have anther question, when Dotspatial create pyramids file itself, a function named DrawToBitmap will spent a lot of time and it seems to save the color value but not the image data value. So if we want to change the color for map, it will spent a long time again.
Thank you for your answer!

**Comment by jany-tenaj** [comment]
delete the mwi and mwh before you add your image

**Comment by wangchao5032** [comment]
@jany-tenaj 
Hum, If I use GdalImageProvider to read image, Dotspatial will check if it exits a GDAL's pyramid file. But if I use GdalRsterProvider, Dotspatial will not check the pyramid file. What is the difference between them?
